pkg/band/band.go

```diff
@@ -154,11 +155,54 @@ func (b Band) FindDownlinkDataRate(dr *ttnpb.DataRate) (ttnpb.DataRateIndex, Dat
 	// NOTE: See notes in FindUplinkDataRate explaining the order of scanning data rates.
 	for i := ttnpb.DataRateIndex_DATA_RATE_15; i >= ttnpb.DataRateIndex_DATA_RATE_0; i-- {
 		bDR, ok := b.DataRates[i]
-		if ok && proto.Equal(bDR.Rate, dr) {
+		if ok && compareDatarates(bDR.Rate, dr, b.StrictCodingRate) {
 			return i, bDR, true
 		}
 	}
 	return 0, DataRate{}, false
 }
 
 func boolPtr(v bool) *bool { return &v }
+
+// compareDatarates checks if two given *ttnpb.DataRate's are equal. If the strict
+// bit is not set the *ttnpb.DataRate_Lora check will skip the `CodingRate`.
+// Make sure to update this check when altering any of the `DataRate` types.
+func compareDatarates(a, b *ttnpb.DataRate, strict bool) bool {
+	switch modA := a.Modulation.(type) {
+	case *ttnpb.DataRate_Lora:
+		switch modB := b.Modulation.(type) {
+		case *ttnpb.DataRate_Lora:
+			if modA.Lora.Bandwidth != modB.Lora.Bandwidth ||
```

Comment:
You can turn these into singular return statements by negating them and removing the `if`.

Reply:
Done, unfortunately the linter is not letting me remove the `default` cases as then it starts complaining about that it could have been an `if` statement even though that is not the case in this instance.

Reply:
Please also apply the negation to the expression itself - do not just wrap the old `if` with it. It becomes hard to parse for other developers if we have too many symbols there, and it is also error prone.

Reply:
My bad, negated the logic and that cleans it up quite nicely.